libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -331,6 +389,8 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 	CodeTransform(
 		m_assembly,
 		m_info,
+		_function.body,
```

Comment:
Because of that we should probably not count the references inside functions, because we will do double work otherwise.

Reply:
ReferencesCounter inherits from ASTWalker, and thus walks the whole AST.

Reply:
@chriseth Yeah, ReferencesCounter always counts that. But, I am not sure if CodeTransform counts the variable references inside a function body (ignoring this function call since it uses a new m_references).

However, I think it is always safe to release smaller reference counts.

Reply:
ReferencesCounter inherits from ASTWalker, and thus walks the whole AST. This includes function bodies.

Reply:
It might be best to create a new `ReferenceCounter` that inherits from the old one but does not visit function definitions.